code/graphics/generic.cpp

```diff
@@ -219,6 +224,29 @@ int generic_anim_stream(generic_anim *ga)
 
 		ga->previous_frame = -1;
 	}
+	else if (ga->type == BM_TYPE_PNG) {
+		if (ga->png.anim == nullptr) {
+			try {
+				ga->png.anim = new apng::apng_ani(ga->filename);
```

Comment:
this may not be trivial; generic_anim is defined as a struct of PODs, "changing it to a class" and adding a non POD member could cause cascading changes throughout the codebase. I'll see how far the rabbit hole goes, if its too deep I'll leave it as a pointer for now.